### ISSUE TITLE
Add to Cart + Options Variation Selector: remove hover styles on disabled pills

### ISSUE THREAD

**Issue by Aljullu** [woothemes/woocommerce#58925]
In the *Add to Cart + Options* variation selector, the disabled pills have hover styles that make them feel clickable.

#### Preparation

1. Go to *Appearance* > *Editor* > *Templates* > *Single Product*.
2. Click on the *Add to Cart with Options* block and update to the blockified version.

#### Reproducing the issue

1. If needed, edit the variations of a variable product to make sure there is a specific combination of attributes where one attribute will be disabled.
2. Visit that product in the frontend.
3. Notice you can hover the pill and the background color changes and the cursor becomes a pointer.

https://github.com/user-attachments/assets/2127bdf8-650d-42eb-aba5-31bdb8be9a48

**Expected result**

When a pill is disabled, the background shouldn't change on hover and the cursor should be the default one.
